### PR TITLE
add owasp dependency check and trivy docker image scan to github actions

### PR DESCRIPTION
**Pull request by ravisingal**

### CODE REVIEW

**Review by tim-mwangi — APPROVED** [review]

File: .github/workflows/dependency-check.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    runs-on: ubuntu-20.04
+    steps:
+       # Set fetch-depth: 0 to fetch commit history and tags for use in version calculation
+      - name: Check out code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: create checksum file
+        uses: hypertrace/github-actions/checksum@main
```
Comment (aaron-steinfeld) [inline .github/workflows/dependency-check.yaml:21]: just a thought, but now that GHA supports composite actions we can probably just create a single shared cache action. They also may have better support for some OOTB caching actions, worth revisiting.

Along the same lines, a single shared dependency check action with all of its setup and reporting should be doable so we can drop in dep check as an independent workflow easily.

File: .github/workflows/pr-build.yml [hunk 2]
```diff
@@ -28,12 +28,32 @@ jobs:
...
-        with: 
+        with:
           args: build dockerBuildImages
 
+      - name: Determine container tag
+        id: tag
+        run: echo tag=$(./gradlew -q printDockerImageDefaultTag | head -1) >> $GITHUB_OUTPUT
+
+      - name: Run Trivy vulnerability scanner
+        uses: aquasecurity/trivy-action@master
```
Comment (aaron-steinfeld) [inline .github/workflows/pr-build.yml:42]: Same thing here - if we want this to be a standard, we can make a shared action with these 20 or so lines.

Comment (ravisingal) [reply .github/workflows/pr-build.yml:42]: Added github action - https://github.com/hypertrace/github-actions/pull/9

File: .github/workflows/dependency-check.yaml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  dependency-check:
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Dependency Check
+        uses: hypertrace/github-actions/dependency-check@dependency-check-trivy-image-scan
```
Comment (aaron-steinfeld) [inline .github/workflows/dependency-check.yaml:15]: pretty! also, use ubuntu 22 unless a reason to use older one

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**